src/mlib/str.h

```diff
@@ -11,6 +11,10 @@
 #include <inttypes.h>
 #include <stdbool.h>
 
+#if defined(MLIB_HAVE_STRINGS_H)
+#include <strings.h> /* For strncasecmp. */
+#endif
```

Comment:
Can we know ahead-of-time whether `<strings.h>` is available? (i.e. by checking the host platform instead of a configure-time check)

Reply:
I am not sure. Checking for `strings.h` is [consistent with libbson](https://github.com/mongodb/mongo-c-driver/blob/671a15154f0dd0e4af3c8df2ac08dfe4acf01795/src/libbson/CMakeLists.txt#L88-L91).